buildSrc/build.gradle.kts

```diff
@@ -4,13 +4,25 @@
 
 plugins {
     `kotlin-dsl`
+    java
+    `java-gradle-plugin`
 }
 
 repositories {
     jcenter()
 }
 
+gradlePlugin {
+    plugins {
+        create("gleanPlugins") {
+            id = "mozilla.components.service.glean-gradle-plugin"
```

Comment:
Strongly consider decoupling the plugin ID from `android-components`.  You probably want this to have an independent name at _some_ time.

Reply:
We did some brainstorming on this today. I think we want to keep this in the android-components namespace because it's the version of the plugin that will evolve in sync with the a-c version of Glean. We will soon have the non-a-c implementation with its own plugin, which may need to evolve in a different direction. If the plugins had the same namespace, I think we're more likely to have a version mismatch between the plugin and the library. (All of this overlap should be very short-lived so I'm more concerned with the possibility of version mismatch between the library and the plugin than I am with the maintenance burden of a fork).

tl;dr: I think it's a feature, not a bug, that this is in the components namespace since it is so closely coupled with the library it supports.